fil_fungible_token/src/token/mod.rs

```diff
@@ -425,11 +429,19 @@ where
                     }
                     .into());
                 }
-                Ok(TransferReturn { from_balance: balance.clone(), to_balance: balance })
+                Ok(TransferReturn {
+                    from_balance: balance.clone(),
+                    to_balance: balance,
+                    hook_return_data: RawBytes::default(),
```

Comment:
Ah, yuk, we don't have the value yet. Token devs are going to screw this up.

Ok, in light of this all of a sudden I like your original design of putting the TransferReturn _inside_ the hook object, and returning it only after the hook is invoked.

Reply:
I've changed it to something close to my original design - it's a bit more tricky behind the scenes but should be much nicer for users of the library.